src/slam_toolbox_common.cpp

```diff
@@ -247,6 +411,18 @@ void SlamToolbox::publishTransformLoop(
 
   rclcpp::Rate r(1.0 / transform_publish_period);
   while (rclcpp::ok()) {
+    {
+      // `get_current_state` is not thread safe in humble
+      // see. <https://github.com/ros2/rclcpp/issues/1746>
+      // fixed in rolling. <https://github.com/ros2/rclcpp/pull/1756>
+      boost::mutex::scoped_lock lock(get_state_mutex_);
+      auto state_id = get_current_state().id();
```

Comment:
This thread should only be running in the active state, where the thread is setup. There's no reason to check this since you setup the `join` in on deactivation to bring down the thread.

Though, this thread needs to look for that signal to join in order to exit. But this isn't necessary